src/main/java/io/cdap/plugin/gcp/bigquery/sqlengine/input/BigQuerySQLEngineInputFormatProvider.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.gcp.bigquery.sqlengine.input;
+
+import io.cdap.plugin.gcp.bigquery.source.BigQueryInputFormatProvider;
+import org.apache.hadoop.conf.Configuration;
+
+/**
+ * Input format provider for BigQuery SQL Engine Pull operations.
+ */
+public class BigQuerySQLEngineInputFormatProvider extends BigQueryInputFormatProvider {
```

Comment:
this is to prevent conflicting exports? if so, would be good to add a comment.

Reply:
No, the reason is that this Input Format Provider overrides `getInputFormatClassName` from `BigQueryInputFormatProvider` in order to return the SQL engine version of the input format: `BigQuerySQLEngineInputFormat`